### PR TITLE
Marketplace price filtering

### PR DESCRIPTION
**Pull request by hwang635**
Implement filtering for the sidebar price menu

- Filtering by 5 hard-coded range options
- Filtering by user-input min + max (takes precedence over range options)
- Toggle apply and unapply for user-input min + max
- Get # of items per price range

### CODE REVIEW

**Comment by curesh** [comment]
This is probably unrelated to this feature but do you know clicking on the item type dropdown in the side bar makes the app crash

File: src/components/Marketplace/MarketplaceScreen.js [hunk 1]
```diff
@@ -94,6 +94,28 @@ export default function MarketplaceScreen() {
...
   const [itemsPerProdType, setProdItems] = useState([]);
 
+  // Manages pallet price filtering
+  const [priceFilters, setPriceFilters] = useState([]);
+  const [appliedRange, setAppliedRange] = useState([]); // [] if no applied min/max
+  const onPriceFilterChange = (newFilters) => {
+    // Parse the min + max #s from each range by splitting + filtering the string
+    const newPriceRanges = [];
+    newFilters.forEach((prices) => {
+      let nums = prices.replace(/\D/g, '#'); // regex expr to replace non-digits w/ #
```
Comment (curesh) [inline src/components/Marketplace/MarketplaceScreen.js:104]: oo fancy

**Review by curesh — COMMENTED** [review]
Wow, I looked through your code for so long and I literally could not find a single thing wrong with it– this code is soo clean. Also this feature is really cool and its nice to see everything working together so well in the marketplace page (this ft along with the last ft you made). Great job!

**Review by curesh — APPROVED** [review]

File: src/components/Marketplace/MarketplaceScreen.js (hunk 1 above)
Comment (hwang635) [reply src/components/Marketplace/MarketplaceScreen.js:104]: thank u from stackoverflow

**Pull request merged**